plugins/cameraserver/src/main/java/edu/wpi/first/shuffleboard/plugin/cameraserver/source/CameraUrlGenerator.java

```diff
@@ -45,31 +47,53 @@ public CameraUrlGenerator(CameraServerSource source) {
     if (frameRate > 0) {
       commands.put("fps", Integer.toString(frameRate));
     }
-    return generateUrls(commands, baseUrls);
+    commands.put("name", source.getName());
+    return generateUrls(commands, baseUrls, source.getName());
   }
 
   @VisibleForTesting
-  static String[] generateUrls(Map<String, String> commands, String[] baseUrls) { // NOPMD varargs instead of array
+  static String[] generateUrls(Map<String, String> commands, String[] baseUrls,
+                               String cameraName) {
     if (baseUrls == null || baseUrls.length == 0) {
       return new String[0];
     }
     if (commands.isEmpty()) {
       return baseUrls;
     } else {
-      return Arrays.stream(baseUrls)
-          .map(url -> url + toHttpParams(commands))
+      var urls =  Arrays.stream(baseUrls)
+          .map(url -> toHttpParams(url, commands))
           .toArray(String[]::new);
+      for (var url : urls) {
+        System.out.println(url);
+      }
+      return urls;
     }
   }
 
   @VisibleForTesting
-  static String toHttpParams(Map<String, String> commands) {
+  static String toHttpParams(String input, Map<String, String> commands) {
     if (commands.isEmpty()) {
-      return "";
+      return input;
     }
-    return commands.entrySet().stream()
-        .map(e -> e.getKey() + "=" + e.getValue())
+    // Special case to remove name from LabVIEW camera
+    if (commands.containsKey("name") && commands.get("name").contains("IMAQdx")) {
+      input = input.replaceAll("\\?name=cam\\d", "");
```

Comment:
Would it be possible to check for `IMAQdx` in the `generateUrls` method when adding the `name` parameter? It seems like that'd be simpler to do than the current approach of unconditionally adding it there, then removing it in this method.

Reply:
We're not removing it in this method. The url that is in the NT publisher already includes `name=camx`, and cscore cannot currently support multiple named parameters. So this removed the one already included, not the extra one that would be added. CS Core might support name encoding in the future, and we would use the name like labview likely, so thats why the new name parameter is added unconditionally.

Reply:
Actually, when we add it we wouldn't need the extra name, so I'll conditional that too.

Reply:
I conditionally added it on the other end as well, as otherwise that name could interfere with a stream that validly uses name.

Reply:
I conditionally added it on the other end as well, as otherwise that name could interfere with a stream that validly uses name.